pkg/virt-controller/watch/vmi.go

```diff
@@ -1125,6 +1126,14 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 				}
 			}
 		}
+		if len(vmispec.NetworksToHotplug(vmi)) > 0 {
+			if err := c.handleDynamicInterfaceRequests(vmi, pod); err != nil {
```

Comment:
Nice. I'll check it out.

Reply:
IIUC, this would work, but I think the current `syncPodAnnotations` code only works when adding new keys.

We need it to also replace old values with newer ones.

The implementation of `syncPodAnnotations` mutates the annotation **if** the key is new **or** if the value of the annotation is different: https://github.com/kubevirt/kubevirt/blob/098332472df0b3740dc68000794a140b5d1984fc/pkg/virt-controller/watch/vmi.go#L359

... but never the less, both clauses will be patched via an add operation: https://github.com/kubevirt/kubevirt/blob/098332472df0b3740dc68000794a140b5d1984fc/pkg/virt-controller/watch/vmi.go#L669

IIUC, when we're updating an existing key's value, we should use a replace operation. 
... and we should probably append a `test` operation first to ensure we're not dropping anything, since we're patching the whole value.

@ormergi

Reply:
Well.. if `syncPodAnnotation()` may introduce issues then please continue with what you already did.
Once we fix `syncPodAnnotation()`to use `replace` and `test`, we can refactor this code to use it.